data/transactions/logic/doc.go

```diff
@@ -0,0 +1,176 @@
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package logic
+
+type stringString struct {
+	a string
+	b string
+}
+
+func stringStringListToMap(they []stringString) map[string]string {
+	out := make(map[string]string, len(they))
+	for _, v := range they {
+		out[v.a] = v.b
+	}
+	return out
+}
+
+// short description of every op
+var opDocList = []stringString{
+	{"err", "Error. Panic immediately. This is primarily a fencepost against accidental zero bytes getting compiled into programs."},
+	{"sha256", "SHA256 hash of value, yields [32]byte"},
+	{"keccak256", "Keccak256 hash of value, yields [32]byte"},
+	{"sha512_256", "SHA512_256 hash of value, yields [32]byte"},
+	{"ed25519verify", "for (data, signature, pubkey) verify the signature of the data against the pubkey => {0 or 1}"},
+	{"rand", "push random uint64 to stack"},
+	{"+", "A plus B. Panic on overflow."},
+	{"-", "A minus B. Panic if B > A."},
+	{"/", "A divided by B. Panic if B == 0."},
+	{"*", "A times B. Panic on overflow."},
+	{"<", "A less than B => {0 or 1}"},
+	{">", "A greater than B => {0 or 1}"},
+	{"<=", "A less than or equal to B => {0 or 1}"},
+	{">=", "A greater than or equal to B => {0 or 1}"},
+	{"&&", "A is not zero and B is not zero => {0 or 1}"},
+	{"||", "A is not zero or B is not zero => {0 or 1}"},
+	{"==", "A is equal to B => {0 or 1}"},
+	{"!=", "A is not equal to B => {0 or 1}"},
+	{"!", "X == 0 yields 1; else 0"},
+	{"len", "yields length of byte value"},
+	{"itob", "converts uint64 to big endian bytes"},
+	{"btoi", "converts bytes as big endian to uint64"},
+	{"%", "A modulo B. Panic if B == 0."},
+	{"|", "A bitwise-or B"},
+	{"&", "A bitwise-and B"},
+	{"^", "A bitwise-xor B"},
+	{"~", "bitwise invert value"},
+	{"mulw", "A times B out to 128-bit long result as low (top) and high uint64 values on the stack"},
+	{"intcblock", "load block of uint64 constants"},
+	{"intc", "push value from uint64 constants to stack by index into constants"},
+	{"intc_0", "push uint64 constant 0 to stack"},
+	{"intc_1", "push uint64 constant 1 to stack"},
+	{"intc_2", "push uint64 constant 2 to stack"},
+	{"intc_3", "push uint64 constant 3 to stack"},
+	{"bytecblock", "load block of byte-array constants"},
+	{"bytec", "push bytes constant to stack by index into constants"},
+	{"bytec_0", "push bytes constant 0 to stack"},
+	{"bytec_1", "push bytes constant 1 to stack"},
+	{"bytec_2", "push bytes constant 2 to stack"},
+	{"bytec_3", "push bytes constant 3 to stack"},
+	{"arg", "push LogicSig.Args[N] value to stack by index"},
+	{"arg_0", "push LogicSig.Args[0] to stack"},
+	{"arg_1", "push LogicSig.Args[1] to stack"},
+	{"arg_2", "push LogicSig.Args[2] to stack"},
+	{"arg_3", "push LogicSig.Args[3] to stack"},
+	{"txn", "push field from current transaction to stack"},
```

Comment:
Suggestion: also autogenerate documentation for txn fields.

Reply:
https://github.com/algobolson/go-algorand/blob/logic-sig-eval/cmd/opdoc/opdoc.go#L209